cmd/thanos/tools_bucket.go

```diff
@@ -483,9 +483,11 @@ func registerBucketDownsample(app extkingpin.AppClause, objStoreConfig *extflag.
 	httpAddr, httpGracePeriod := extkingpin.RegisterHTTPFlags(cmd)
 	dataDir := cmd.Flag("data-dir", "Data directory in which to cache blocks and process downsamplings.").
 		Default("./data").String()
+	hashFunc := cmd.Flag("hash-func", "Specify which hash function to use when calculating the hashes of produced files. If no function has been specified, it does not happen. This permits avoiding downloading some files twice albeit at some performance cost. Possible values are: \"\", \"SHA256\".").
```

Comment:
Because it doesn't make sense to add it for both writers _and_ readers. It would not mean anything for components that only read and would only raise confusion e.g. `bucket ls`. As for separating this into a helper or something - maybe. I had an idea that we could separate the uploading part from `Shipper` into a separate struct/package and use that inside of shipper + here. But that is a completely separate story that I think we can do later given that such a refactoring does not impact the end user at all.